contexts/TokenContext.tsx

```diff
@@ -0,0 +1,53 @@
+import useNotifications from "@/hooks/useNotifications";
+import useTokenBalance from "@/hooks/useTokenBalance";
+import { useTokenList, TokenListItem } from "@/hooks/useTokenList";
+import { BigNumber } from "@ethersproject/bignumber";
+import React, { ReactNode, createContext, useContext } from "react";
+
+type TokenListContextItem = TokenListItem & { balance: BigNumber | null };
+
+export type TokenContextType = {
+  tokenList: TokenListContextItem[];
+  isLoading: boolean;
+};
+const initialContext = {
+  isLoading: false,
+  tokenList: [],
+};
+
+const TokenContext = createContext<TokenContextType>(initialContext);
```

Comment:
TokenContext  has the data from useTokenList with the balance value added to it. It stores the value in a higher level state that can be accessed in lower level components.

We need it for the balance info and the regular hooks don't persist data across components. So, if we need to use token data in multiple parts of the app we would be fetching multiple times. That being said we could combine some of these and even remove the context, but then some of the logic would have multiple responsibilities

Reply:
> We need it for the balance info and the regular hooks don't persist data across components. So, if we need to use token data in multiple parts of the app we would be fetching multiple times.

Wdym by this exactly? My understanding was that `useSWR` handles the http requests to persist the response / avoid duplicate fetches when the hook is used in multiple components, by fetch every 3 seconds and caching results instead (my understanding could be wrong)

Reply:
I think your understanding is correct, but I think SWR has some internal cache in memory that maintains the data.

What I mean by the statement above is instead of sharing data through passing props down through the component tree we could use context which allows us to skip components in the component tree and share state. Regular hooks will only scope data to the component they are being used in unless they are using context or some other state management solution.

Reply:
That being said, it may be simpler to move balances outside of this context because it is seeming more and more likely it will potentially change frequently

Reply:
Got it, appreciate the explanation! Deferring to you in case we need an issue to track that, or if the change will come naturally from other changes we make. Feel free to resolve this comment yourself.